proxy/apifilterproxy/filters/auth/rancher_token_validate_filter.go

```diff
@@ -0,0 +1,277 @@
+package auth
+
+import (
+	"encoding/json"
+	"errors"
+	log "github.com/Sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/filters"
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/model"
+)
+
+const (
+	name = "authTokenValidator"
+)
+
+//AuthorizeData is for the JSON output
+type AuthorizeData struct {
+	Message string `json:"message,omitempty"`
+}
+
+//MessageData is for the JSON output
+type MessageData struct {
+	Data []interface{} `json:"data,omitempty"`
+}
+
+func init() {
+	if len(os.Getenv("PROXY_CATTLE_ADDRESS")) == 0 {
+		log.Infof("PROXY_CATTLE_ADDRESS is not set, skipping init of %v API filter", name)
+		return
+	}
+	tokenFilter := &TokenValidationFilter{}
+	tokenFilter.rancherURL = "http://" + os.Getenv("PROXY_CATTLE_ADDRESS")
+
+	if err := filters.RegisterAPIFilter(name, tokenFilter); err != nil {
+		log.Fatalf("Could not register %s filter", name)
+	}
+
+	log.Infof("Configured %s API filter", tokenFilter.GetName())
+
+}
+
+type TokenValidationFilter struct {
+	rancherURL string
+}
+
+func (*TokenValidationFilter) GetName() string {
+	return name
+}
+
+func (f *TokenValidationFilter) ProcessFilter(filter model.FilterData, input model.APIRequestData) (model.APIRequestData, error) {
+	output := model.APIRequestData{}
+
+	envid := input.EnvID
+
+	log.Debugf("Request => %v", input)
+
+	var cookie []string
+	if len(input.Headers["Cookie"]) >= 1 {
+		cookie = input.Headers["Cookie"]
+	} else {
+		log.Debugf("No Cookie found.")
+		output.Status = http.StatusOK
+		return output, nil
+	}
+	var cookieString string
+	if len(cookie) >= 1 {
+		for i := range cookie {
+			if strings.Contains(cookie[i], "token") {
+				cookieString = cookie[i]
+			}
+		}
+	} else {
+		log.Debugf("No token found in cookie.")
+		output.Status = http.StatusOK
+		return output, nil
+	}
+
+	tokens := strings.Split(cookieString, ";")
+	tokenValue := ""
+	if len(tokens) >= 1 {
+		for i := range tokens {
+			if strings.Contains(tokens[i], "token") {
+				if len(strings.Split(tokens[i], "=")) > 1 {
+					tokenValue = strings.Split(tokens[i], "=")[1]
+				}
+			}
+
+		}
+	} else {
+		log.Errorf("No token found")
+		output.Status = http.StatusOK
+		return output, nil
+	}
+	if tokenValue == "" {
+		log.Errorf("No token found")
+		output.Status = http.StatusOK
+		return output, nil
```

Comment:
do we ever return not nil error in this function? If not - and the only thing we do is create ok vs non ok output status - may be it would make sense to remove the error from the list of returned arguments?